R/cal-binary.R

```diff
@@ -0,0 +1,674 @@
+#' @param include_ribbon Flag that indicates if the ribbon layer is to be
+#' included. Defaults to `TRUE`
+#' @param include_rug Flag that indicates if the Rug layer is to be included.
+#' Defaults to `TRUE`
+#' @seealso These functions depend on tables built by the following corresponding
+#' functions: [cal_binary_table_breaks()], [cal_binary_table_logistic()], and
+#' [cal_binary_table_windowed()]
+#' @examples
+#'
+#' library(ggplot2)
+#'
+#' cal_binary_plot_breaks(
+#'   segment_logistic,
+#'   Class,
+#'   .pred_good
+#' )
+#'
+#' cal_binary_plot_logistic(
+#'   segment_logistic,
+#'   Class,
+#'   .pred_good
+#' )
+#'
+#' cal_binary_plot_windowed(
+#'   segment_logistic,
+#'   Class,
+#'   .pred_good
+#' )
+#'
+#' @export
+cal_binary_plot_breaks <- function(.data,
+                                   truth = NULL,
+                                   estimate = NULL,
+                                   num_breaks = 10,
+                                   conf_level = 0.90,
+                                   include_ribbon = TRUE,
+                                   include_rug = TRUE,
+                                   event_level = c("first", "second")) {
+  UseMethod("cal_binary_plot_breaks")
+}
+
+cal_binary_plot_breaks_impl <- function(.data,
+                                        truth = NULL,
+                                        estimate = NULL,
+                                        num_breaks = 10,
+                                        conf_level = 0.90,
+                                        include_ribbon = TRUE,
+                                        include_rug = TRUE,
+                                        event_level = c("first", "second")) {
+  truth <- enquo(truth)
+  estimate <- enquo(estimate)
+
+  assert_truth_two_levels(.data, !!truth)
+
+  prob_tbl <- cal_binary_table_breaks(
+    .data = .data,
+    truth = !!truth,
+    estimate = !!estimate,
+    num_breaks = num_breaks,
+    conf_level = conf_level,
+    event_level = event_level
+  )
+
+  binary_plot_impl(
+    prob_tbl, predicted_midpoint, event_rate,
+    .data, !!truth, !!estimate,
+    "Predicted Midpoint", "Event Rate",
+    NULL, include_ribbon, include_rug, TRUE
+  )
+}
+
+#' @export
+cal_binary_plot_breaks.data.frame <- cal_binary_plot_breaks_impl
+
+#' @export
+cal_binary_plot_breaks.tune_results <- function(.data,
+                                                truth = NULL,
+                                                estimate = NULL,
+                                                num_breaks = 10,
+                                                conf_level = 0.90,
+                                                include_ribbon = TRUE,
+                                                include_rug = TRUE,
+                                                event_level = c("first", "second")) {
+  rs <- tune::collect_predictions(.data, summarize = TRUE)
```

Comment:
I will add a function later to check these results. We ned to make sure we give them a good error message when they did not use `save_pred = TRUE`. There is also some things related to racing methods that I'll need to check for.